storage.tf

```diff
@@ -19,7 +19,7 @@ locals {
           capacity       = volume.capacity
           vsi_name       = "${var.prefix}-${(subnet) * (var.vsi_per_subnet) + count + 1}"
           iops           = volume.iops
-          encryption_key = volume.encryption_key
+          encryption_key = var.existing_kms_instance_guid != null ? var.boot_volume_encryption_key : volume.encryption_key
```

Comment:
`existing_kms_instance_guid` is not the right variable to use here. `existing_kms_instance_guid` is only needed if you want to create the auth policy. In ICS modules we use a variable called `kms_encryption_enabled`. Does anything like this exist in this module already? If not, I think we need it